From e9ac56944c12e62ff2ab25a35ba3fa65d2d70b18 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 7 Jun 2017 17:48:02 +0200
Subject: patches: fix upgrade

Closes  #3386.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/patches.go | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lxd/patches.go b/lxd/patches.go
index e3db87b6..525dd68b 100644
--- a/lxd/patches.go
+++ b/lxd/patches.go
@@ -43,6 +43,7 @@ var patches = []patch{
 	{name: "storage_api_update_storage_configs", run: patchStorageApiUpdateStorageConfigs},
 	{name: "storage_api_lxd_on_btrfs", run: patchStorageApiLxdOnBtrfs},
 	{name: "storage_api_lvm_detect_lv_size", run: patchStorageApiDetectLVSize},
+	{name: "storage_api_insert_zfs_driver", run: patchStorageApiInsertZfsDriver},
 }
 
 type patch struct {
@@ -2237,6 +2238,15 @@ func patchStorageApiDetectLVSize(name string, d *Daemon) error {
 	return nil
 }
 
+func patchStorageApiInsertZfsDriver(name string, d *Daemon) error {
+	_, err := dbExec(d.db, "UPDATE storage_pools SET driver='zfs', description='' WHERE driver=''")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // Patches end here
 
 // Here are a couple of legacy patches that were originally in
